packages/store-sync/src/watchLogs.ts

```diff
@@ -30,56 +30,19 @@ export function watchLogs({ url, address, fromBlock }: WatchLogsInput): WatchLog
 
   let resumeBlock = fromBlock;
 
-  let pingTimer: ReturnType<typeof setTimeout> | undefined = undefined;
-  // how often to ping to keep socket alive
-  const pingInterval = 3000;
-  // how long to wait for ping response before we attempt to reconnect
-  const pingTimeout = 5000;
-
   const logs$ = new Observable<StorageAdapterBlock>((subscriber) => {
-    debug("logs$ subscribed");
+    debug("wiresaw_watchLogs subscribed, starting from", fromBlock);
 
     let client: SocketRpcClient<WebSocket>;
 
     async function setupClient(): Promise<void> {
-      debug("setupClient called");
-
-      // Buffer the live logs received until the gap from `startBlock` to `currentBlock` is closed
+      // Buffer the live logs received until the gap from `fromBlock` to `currentBlock` is closed
       let caughtUp = false;
       const logBuffer: StoreEventsLog[] = [];
 
-      client = await getWebSocketRpcClient(url, { keepAlive: false });
-      debug("got websocket rpc client");
-
-      async function ping(): Promise<void> {
-        try {
-          debug("pinging socket");
-          await client.requestAsync({ body: { method: "net_version" }, timeout: pingTimeout });
-        } catch (error) {
-          debug("ping failed, closing...", error);
-          client.close();
-        }
-      }
-
-      function schedulePing(): void {
-        debug("scheduling next ping");
-        pingTimer = setTimeout(() => ping().then(schedulePing), pingInterval);
-      }
-
-      schedulePing();
-
-      client.socket.addEventListener("error", (error) => {
-        debug("socket error, closing...", error);
-        client.close();
-      });
-
-      client.socket.addEventListener("close", async () => {
-        debug("socket close, setting up new client...");
-        clearTimeout(pingTimer);
-        setupClient().catch((error) => {
-          debug("error trying to setup new client", error);
-          subscriber.error(error);
-        });
+      client = await getWebSocketRpcClient(getUncachedUrl(url), {
+        keepAlive: true,
+        reconnect: { attempts: 100, delay: 1_000 },
```

Comment:
is it intentional that attempts is really high? won't this mean retrying ~forever because of how the delay scales/backs off?

Reply:
yeah that's intentional, the idea is that we're just using the latest block stream while this is disconnected or not returning valid logs (ie because wiresaw is down), but we want to keep retrying to automatically recover once it becomes available again